iwyu_path_util.cc

```diff
@@ -149,6 +149,9 @@ string NormalizeFilePath(const string& path) {
   string result = CanonicalizeFilePath(path);
   while (StripLeft(&result, "./")) {
   }
+  // Don't use StripRight as we might have 'foo/../' here
```

Comment:
This comment is either wrong or needs more elaboration :-)

It looks like you're checking for `"/./"` but stripping only `"./"` -- is that why `StripRight` is not possible?